system-design/specification/applications/application-registry.md

```diff
@@ -0,0 +1,259 @@
+# Application Registry
+
+The interactions of an `Application Registry` are described [here](../../concepts/applications/application-registry.md) from a conceptual view. This section formally specifies the API of the `Application Registry` and the exchange of an [Application Package](../../concepts/applications/application-package.md), defined through an [Application Description](../../specification/applications/application-description.md) file, from an Application Developer to the [Workload Fleet Manager](../../personas-and-definitions/technical-lexicon.md#workload-fleet-manager) (WFM). The `Application Registry` is designed as an `OCI Registry`, i.e., it offers an API compliant with the [OCI Registry API (v1.1.0)](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md) (called here the "OCI_spec") for digital artifact distribution. This way, the Application Registry hosts the parts of an [Application Package](../../concepts/applications/application-package.md) in the form of `blobs` and uses `image manifests` according to [OCI Image specification](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md) to list a set of `layers`, each pointing at a blob.
+
+## Overview of API Endpoints
+
+The WFM MUST interact with the Application Registry compliant to the OCI Registry API endpoints defined in the [OCI_spec](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md):
+
+* Tags: `/v2/{name}/tags/list` for listing available versions of an Application Package
+* Manifest: `/v2/{name}/manifests/{reference}` for retrieving an image manifest (identifed through the `{reference}`) that lists layers, which represent the parts of the Application Package.
+* Blob: `/v2/{name}/blobs/{digest}` for downloading parts of the Application Package
+
+`{name}` is the namespace of the repository, which needs to be directly communicted by the App Developer to the WFM vendor. It could be for example a combination of the organization's and application's name.
+
+Further details on how to use these API endpoints are specified below [towards App Developer](#uploading-an-application-package) and [towards WFM](#retrieving-an-application-package).
+
+## Authentication, Authorization & Security
+Margo recommends the use of an Authentication Service for the interactions between the WFM and Application Registry as conceptually described [here](../../concepts/applications/application-registry.md), e.g., implemented using OAuth 2.0. This involves the following workflow:
+
+* WFM obtains credentials during onboarding
+* WFM requests a token from an Authentication Service
+* WFM uses the token for subsequent API calls to the Application Registry
+* Application Registry validates the token and enforces access control
+
+Thereby, all communications must use TLS 1.3+ to ensure transport security.
```

Comment:
Do we need to fix it here? I'd rather refer to centrally specified minimum security standards in the spec.